### ISSUE TITLE
Warn when `distinct()` is used outside of aggregated functions

### ISSUE THREAD

**Issue by CaselIT** [sqlalchemy/sqlalchemy#11526]
Discussed in #11518

The alternatives could be:
- at compile time: you could do it at compile time by passing a kw flag that "distinct" looks for , indicating the appropriate context. if flag not there, boom. would have to find a way to make it OK for str() compile -- https://github.com/sqlalchemy/sqlalchemy/discussions/11518#discussioncomment-9840468
- at coercion time: does "distinct()" only go inside of a SQL function ? That's all we'd have to look for. we can put "distinct()" into a special coercion category based on the roles, like FunctionDistinctRole -- https://github.com/sqlalchemy/sqlalchemy/discussions/11518#discussioncomment-9841617

the second option is likely better since it will give better feedback to the user (since it will happen at statement creation time, instead of when it will be compiled)

**Comment by koenlists** [comment]
Hi @CaselIT, is it oke if I try to pick this up?

**Comment by CaselIT** [comment]
Sure, feel free to try this

**Comment by bekapono** [comment]
@koenlists  @CaselIT 

Hey, checking in to see if this is still open, and can I pick this up?

**Comment by CaselIT** [comment]
yes, I don't think there is any progress on this at the moment. if you want to give it a try feel free

**Comment by sqla-tester** [comment]
**bekapono** has proposed a fix for this issue in the **main** branch:

**Coercion warning feedback for unary distinct outside aggregate function** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/6632
